data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetAttestations.java

```diff
@@ -63,16 +93,28 @@ public GetAttestations(final DataProvider dataProvider, final JsonProvider jsonP
         @OpenApiResponse(status = RES_INTERNAL_ERROR),
       })
   @Override
-  public void handle(final Context ctx) throws Exception {
-    Map<String, List<String>> queryParamMap = ctx.queryParamMap();
-    Optional<UInt64> maybeSlot =
-        SingleQueryParameterUtils.getParameterValueAsUInt64IfPresent(queryParamMap, SLOT);
-    Optional<UInt64> maybeCommitteeIndex =
-        SingleQueryParameterUtils.getParameterValueAsUInt64IfPresent(
-            ctx.queryParamMap(), COMMITTEE_INDEX);
+  public void handle(@NotNull final Context ctx) throws Exception {
     ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
-    List<Attestation> attestations =
-        nodeDataProvider.getAttestations(maybeSlot, maybeCommitteeIndex);
-    ctx.json(jsonProvider.objectToJSON(new GetAttestationsResponse(attestations)));
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    Optional<UInt64> slot = request.getOptionalQueryParameter(SLOT_PARAMETER);
+    Optional<UInt64> committeeIndex = request.getOptionalQueryParameter(COMMITTEE_INDEX_PARAMETER);
+
+    List<Attestation> attestations = nodeDataProvider.getAttestations(slot, committeeIndex);
```

Comment:
i'd probably not create this variable, just 
`request.respondOk(nodeDataProvider.getAttestations(slot, committeeIndex));`